static/js/AddToSourceSheet.jsx

```diff
@@ -120,11 +120,32 @@ class AddToSourceSheetBox extends Component {
           source[language.slice(0,2)] = selectedWords;
         }
       }
-      let postData = {source: JSON.stringify(source)};
-      if (this.props.note) {
-        postData.note = this.props.note;
+      if (this.validateRefs(source.refs)) {
+        let postData = {source: JSON.stringify(source)};
+        if (this.props.note) {
+          postData.note = this.props.note;
+        }
+        $.post(url, postData, this.confirmAdd);
+      }
+  }
+  validateRefs(refs) {
+    // validate refs have no images before posting them to sheet
+    let foundImage = false;
+    refs.forEach(r => {
+      if (!foundImage) {
+        const ref = Sefaria.getRefFromCache(r);
+        if (ref) {
+          foundImage = Sefaria.isImage(ref.he) || Sefaria.isImage(ref.text);
+        }
       }
-      $.post(url, postData, this.confirmAdd);
+    });
+    if (foundImage) {
+      alert("One of the sources you tried to add contains an image.  Unfortunately, we do not currently support adding images to source sheets.");
```

Comment:
@stevekaplan123 I don't think we need to be apologetic! It can simply be `We do not currently...`